plasmapy/tests/test_requirements.py

```diff
@@ -0,0 +1,140 @@
+
+    Parameters
+    ----------
+    prefix : str
+        The prefix to a filename, such as ``"build"`` for ``build.txt``.
+
+    requirements_directory : str
+        The path to the directory containing the requirements file.
+
+    Returns
+    -------
+    list of str
+        A `list` containing the lines of the requirements file,
+        excluding lines that do not start with an alphabetic character
+        (like comments).
+    """
+    filename = f"{requirements_directory}/{prefix}.txt"
+    with open(filename) as file:
+        lines_of_file = file.readlines()
+    return {line.strip() for line in lines_of_file if line[0].isalpha()}
+
+
+def get_requirements_from_txt() -> Dict[str, Set[str]]:
+    """
+    Get the requirements from the ``.txt`` files in the ``requirements``
+    directory.
+    """
+    return {
+        prefix: read_requirements_txt_file(prefix, requirements_directory)
+        for prefix in requirements_prefixes
+    }
+
+
+def get_requirements_from_setup_cfg() -> Dict[str, Set[str]]:
+    """Get the requirements that are contained in ``setup.cfg``."""
+    configuration = setuptools.config.read_configuration(f"{base_directory}/setup.cfg")
+    return {
+        "docs": set(configuration["options"]["extras_require"]["docs"]),
+        "extras": set(configuration["options"]["extras_require"]["extras"]),
+        "install": set(configuration["options"]["install_requires"]),
+        "tests": set(configuration["options"]["extras_require"]["tests"]),
+    }
+
+
+def get_requirements_from_pyproject_toml() -> Dict[str, Set[str]]:
+    """Get the requirements that are contained in pyproject.toml."""
+    pyproject_toml = toml.load(f"{base_directory}/pyproject.toml")
+    return {"build": set(pyproject_toml["build-system"]["requires"])}
+
+
+requirements_from_txt = get_requirements_from_txt()
+requirements_from_pyproject_toml = get_requirements_from_pyproject_toml()
+requirements_from_setup_cfg = get_requirements_from_setup_cfg()
+
+requirements_table = [
+    (
+        "build",
+        requirements_from_txt["build"],
+        "requirements/build.txt",
+        requirements_from_pyproject_toml["build"],
+        "pyproject.toml",
+    ),
+    (
+        "install",
+        requirements_from_txt["install"],
+        "requirements/install.txt",
+        requirements_from_setup_cfg["install"],
+        "setup.cfg",
+    ),
+    (
+        "extras",
+        requirements_from_txt["extras"],
+        "requirements/extras.txt",
+        requirements_from_setup_cfg["extras"],
+        "setup.cfg",
+    ),
+    (
+        "docs",
+        requirements_from_txt["docs"],
+        "requirements/docs.txt",
+        requirements_from_setup_cfg["docs"] - requirements_from_txt["extras"],
+        "setup.cfg",
+    ),
+    (
+        "tests",
+        requirements_from_txt["tests"],
+        "requirements/tests.txt",
+        requirements_from_setup_cfg["tests"] - requirements_from_txt["extras"],
+        "setup.cfg",
+    ),
+]
```

Comment:
I was thinking about refactoring this to reduce the number of parameters used in the test parametrization, but it'd be a lot of effort for not much gain in readability/maintainability.  In particular, this would be harder because of the special cases that the `tests` and `docs` requirements include the `extras` requirement in `setup.cfg`.

Reply:
What about just compressing the first three lines to, for the last test case, just `"tests"`, and getting the other two lines in the function body?